openfoodfoundation/openfoodnetwork
# [Cookies] Adds matomo optout iframe to cookies policy page

#### What? Why?

Closes #2518 

#### What should we test?
##### 1. Cookies policy page
Description: Cookies policy page should be retested.

##### 2. Test opt-out configuration
Description: The opt-out iframe shows up IF matomo content is configured to display on cookies policy page AND IF there is a matomo url configured in the matomo settings.
Steps:
- Go to backoffice, General Settings, check "Display Matomo section on cookies policy page", save
- Go to backoffice, Matomo Settings, paste some URL on Matomo URL, save
- Go to cookies policy page and verify there is a opt.out checkbox you can use
- Go to backoffice, Matomo Settings
- Delete content from Matomo URL, save
- Go to cookies policy page and verify there is no opt-out checkbox

##### 3. Test optout iframe in the cookies policy page
Description: If Matomo is configured, the user should be able to click on the optout checkbox in the cookies policy page. After this, the user behaviour should not be tracked in matomo.
Steps:
- Go to the backoffice and paste some url on the matomo url in matomo settings
- Go to cookies policy page and check the matomo optout checkbox
- Using your browser dev tools, in Application/Cookies, verify that there is a new cookie set piwik_ignore
- I checked and I couldn't find any other way of verifying this in Matomo

#### Release notes
Changelog Category: Added
If Matomo integration is configured, the cookies policy page now includes a optout checkbox that enables the user to optout from being tracked with matomo.

#### Documentation updates
See https://github.com/openfoodfoundation/openfoodnetwork/wiki/Matomo-Integration
